mentoring/public/js/mentoring_assessment_view.js

```diff
@@ -294,11 +298,25 @@ function MentoringAssessmentView(runtime, element, mentoring) {
         if (submitXHR) {
             submitXHR.abort();
         }
-        submitXHR = $.post(handlerUrl, JSON.stringify(data)).success(callback);
+        submitXHR = $.ajax({
+            'type': 'POST',
+            'url': handlerUrl,
+            'data': JSON.stringify(data),
+            'complete': complete_callback,
+            'success' : success_callback
+        });
     }
 
     function submit() {
-        calculate_results('submit', handleSubmitResults)
+        submitDOM.attr('disabled','disabled');
+        enable_after = function (jxhr, status) {
+            // If status==success button was re-enabled by the  handleSubmitResults
+            // which launches earlier.
+            if (status != 'success') {
```

Comment:
I'm not sure, from the jQuery documentation it wern't completly clear whether success and error cover all possible states. But it seems that they do indeed, so I'll change to error status.